cattle/plugins/docker/storage.py

```diff
@@ -175,10 +175,43 @@ def _get_volume_storage_pool_map_data(self, obj):
         }
 
     def _is_volume_active(self, volume, storage_pool):
+        if hasattr(volume.data, 'fields'):
+            if not hasattr(volume.data.fields, 'volumeDriver'):
+                return True
+        else:
+            return True
+        if volume.data.fields['volumeDriver'] is None:
+            return True
+        if volume.data.fields['volumeDriver'] == "":
+            return True
+        if volume.name is None or volume.name == "":
+            return True
+        try:
+            docker_client().inspect_volume(volume.name)
+        except:
+            return False
         return True
 
+    def _do_volume_activate(self, volume, storage_pool, progress):
+        driver = ""
+        driverOpts = None
+        if hasattr(volume.data, 'fields'):
+            if hasattr(volume.data.fields, 'volumeDriver'):
+                driver = volume.data.fields['volumeDriver']
+            if hasattr(volume.data.fields, 'volumeDriverOpts'):
+                driverOpts = volume.data.fields['volumeDriverOpts']
+        docker_client().create_volume(volume.name, driver, driverOpts)
+
     def _is_volume_inactive(self, volume, storage_pool):
-        return True
+        if not hasattr(volume, 'name'):
+            return True
+        if volume.name is None or volume.name == "":
+            return True
+        try:
+            docker_client().inspect_volume(volume.name)
```

Comment:
At this time can we pass the 1.21 version to create the client.  Create another config setting for storage_api_version, or something like that.